### ISSUE TITLE
The OAT Gate does not support parameters

### ISSUE THREAD

**Issue by fretchen** [Qiskit-Extensions/qiskit-cold-atom#22]
It would seem that the OAT gate does not support the binding of parameters for the moment, which is rather unfortunate. I would have liked to write:

```
qc = QuantumCircuit(1)

alpha = Parameter("α")
beta = Parameter("β")
gamma = Parameter("γ")

qc.ly(np.pi/2, 0)
qc.append(OATGate(chi=gamma, delta=0, omega=beta), qargs=[0])
qc.lx(np.pi, 0)
qc.append(OATGate(chi=gamma, delta=0, omega=beta), qargs=[0])
qc.lx(alpha, 0)
qc.measure_all()
```

However, once I attempt to assign the parameters:

```
circuit = qc.assign_parameters(
        {
            alpha: np.pi/2,
            beta: 0,
            gamma: 1,
        }, inplace=False
    )
job = backend.run(circuit, shots = 1000, spin=Nat/2, seed=14)
```

I obtain the following bug:
`Unsupported operand type(s) for *: 'SpinOp' and 'ParameterExpression' `

**Comment by fretchen** [comment]
IMHO this should be fixed in PR #17

**Comment by LaurinFischer** [comment]
This is indeed fixed through PR #17, thanks!
